In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
# from sklearn.preprocessing

In [37]:
# rawData = pd.read_csv('/content/forestfires.csv')
rawData = pd.read_csv('forestfires.csv')
rawData.head()

month  day  FFMC   DMC     DC  ISI  temp  RH  wind  rain  ...  monthfeb  \
0   mar  fri  86.2  26.2   94.3  5.1   8.2  51   6.7   0.0  ...         0   
1   oct  tue  90.6  35.4  669.1  6.7  18.0  33   0.9   0.0  ...         0   
2   oct  sat  90.6  43.7  686.9  6.7  14.6  33   1.3   0.0  ...         0   
3   mar  fri  91.7  33.3   77.5  9.0   8.3  97   4.0   0.2  ...         0   
4   mar  sun  89.3  51.3  102.2  9.6  11.4  99   1.8   0.0  ...         0   

   monthjan  monthjul  monthjun  monthmar  monthmay  monthnov  monthoct  \
0         0         0         0         1         0         0         0   
1         0         0         0         0         0         0         1   
2         0         0         0         0         0         0         1   
3         0         0         0         1         0         0         0   
4         0         0         0         1         0         0         0   

   monthsep  size_category  
0         0          small  
1         0          small  
2         0          small  
3         0          small  
4         0          small  

[5 rows x 31 columns]

In [38]:
rawData = rawData.iloc[:,2:11].copy(deep=True)
rawData

FFMC    DMC     DC   ISI  temp  RH  wind  rain   area
0    86.2   26.2   94.3   5.1   8.2  51   6.7   0.0   0.00
1    90.6   35.4  669.1   6.7  18.0  33   0.9   0.0   0.00
2    90.6   43.7  686.9   6.7  14.6  33   1.3   0.0   0.00
3    91.7   33.3   77.5   9.0   8.3  97   4.0   0.2   0.00
4    89.3   51.3  102.2   9.6  11.4  99   1.8   0.0   0.00
..    ...    ...    ...   ...   ...  ..   ...   ...    ...
512  81.6   56.7  665.6   1.9  27.8  32   2.7   0.0   6.44
513  81.6   56.7  665.6   1.9  21.9  71   5.8   0.0  54.29
514  81.6   56.7  665.6   1.9  21.2  70   6.7   0.0  11.16
515  94.4  146.0  614.7  11.3  25.6  42   4.0   0.0   0.00
516  79.5    3.0  106.7   1.1  11.8  31   4.5   0.0   0.00

[517 rows x 9 columns]

In [4]:
rawData.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 517 entries, 0 to 516
Data columns (total 9 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   FFMC    517 non-null    float64
 1   DMC     517 non-null    float64
 2   DC      517 non-null    float64
 3   ISI     517 non-null    float64
 4   temp    517 non-null    float64
 5   RH      517 non-null    int64  
 6   wind    517 non-null    float64
 7   rain    517 non-null    float64
 8   area    517 non-null    float64
dtypes: float64(8), int64(1)
memory usage: 36.5 KB


In [5]:
from sklearn.preprocessing import StandardScaler,MinMaxScaler

In [39]:
Scaler = StandardScaler()
# Mscaler = MinMaxScaler()

In [7]:
# sns.pairplot(rawData)

In [8]:
from keras.models import Sequential
from keras.layers import Dense

In [45]:
Scaled_Data = pd.DataFrame(Scaler.fit_transform(rawData.iloc[:,:9]),columns=rawData.columns[:9])
# X = pd.DataFrame(Mscaler.fit_transform(rawData.iloc[:,:8]),columns=rawData.columns[:8])
X = Scaled_Data.iloc[:,:8]
X

FFMC       DMC        DC       ISI      temp        RH      wind  \
0   -0.805959 -1.323326 -1.830477 -0.860946 -1.842640  0.411724  1.498614   
1   -0.008102 -1.179541  0.488891 -0.509688 -0.153278 -0.692456 -1.741756   
2   -0.008102 -1.049822  0.560715 -0.509688 -0.739383 -0.692456 -1.518282   
3    0.191362 -1.212361 -1.898266 -0.004756 -1.825402  3.233519 -0.009834   
4   -0.243833 -0.931043 -1.798600  0.126966 -1.291012  3.356206 -1.238940   
..        ...       ...       ...       ...       ...       ...       ...   
512 -1.640083 -0.846648  0.474768 -1.563460  1.536084 -0.753800 -0.736124   
513 -1.640083 -0.846648  0.474768 -1.563460  0.519019  1.638592  0.995798   
514 -1.640083 -0.846648  0.474768 -1.563460  0.398350  1.577248  1.498614   
515  0.680957  0.549003  0.269382  0.500176  1.156839 -0.140366 -0.009834   
516 -2.020879 -1.685913 -1.780442 -1.739089 -1.222058 -0.815143  0.269509   

         rain  
0   -0.073268  
1   -0.073268  
2   -0.073268  
3    0.603155  
4   -0.073268  
..        ...  
512 -0.073268  
513 -0.073268  
514 -0.073268  
515 -0.073268  
516 -0.073268  

[517 rows x 8 columns]

In [46]:
X.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 517 entries, 0 to 516
Data columns (total 8 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   FFMC    517 non-null    float64
 1   DMC     517 non-null    float64
 2   DC      517 non-null    float64
 3   ISI     517 non-null    float64
 4   temp    517 non-null    float64
 5   RH      517 non-null    float64
 6   wind    517 non-null    float64
 7   rain    517 non-null    float64
dtypes: float64(8)
memory usage: 32.4 KB


In [47]:
y = Scaled_Data.area
y

0     -0.202020
1     -0.202020
2     -0.202020
3     -0.202020
4     -0.202020
         ...   
512   -0.100753
513    0.651674
514   -0.026532
515   -0.202020
516   -0.202020
Name: area, Length: 517, dtype: float64

In [48]:
X = np.array(X)
y = np.array(y)
type(X),type(y)

(numpy.ndarray, numpy.ndarray)

In [90]:
model = Sequential()
model.add(Dense(19, input_dim=8,  activation='relu'))
model.add(Dense(32, activation = 'relu'))
model.add(Dense(16, activation = 'relu'))
model.add(Dense(1, activation = 'relu'))

In [91]:
model.compile(loss='mse', optimizer='adam', metrics='mean_absolute_error' ) 

In [92]:
model.fit(X, y, validation_split=0.33, epochs=250, batch_size=20)

Epoch 1/250
18/18 [==============================] - 3s 42ms/step - loss: 1.7058 - mean_absolute_error: 0.3241 - val_loss: 1.0210 - val_mean_absolute_error: 0.3147
Epoch 2/250
18/18 [==============================] - 0s 8ms/step - loss: 1.4821 - mean_absolute_error: 0.3174 - val_loss: 1.0210 - val_mean_absolute_error: 0.3147
Epoch 3/250
18/18 [==============================] - 0s 9ms/step - loss: 0.5248 - mean_absolute_error: 0.2469 - val_loss: 1.0210 - val_mean_absolute_error: 0.3147
Epoch 4/250
18/18 [==============================] - 0s 3ms/step - loss: 0.2927 - mean_absolute_error: 0.2232 - val_loss: 1.0210 - val_mean_absolute_error: 0.3147
Epoch 5/250
18/18 [==============================] - 0s 5ms/step - loss: 0.8008 - mean_absolute_error: 0.2639 - val_loss: 1.0210 - val_mean_absolute_error: 0.3147
Epoch 6/250
18/18 [==============================] - 0s 7ms/step - loss: 1.0822 - mean_absolute_error: 0.3130 - val_loss: 1.0210 - val_mean_absolute_error: 0.3147
Epoch 7/250
18/18 [==

Epoch 51/250
18/18 [==============================] - 0s 6ms/step - loss: 0.7719 - mean_absolute_error: 0.2534 - val_loss: 1.0210 - val_mean_absolute_error: 0.3147
Epoch 52/250
18/18 [==============================] - 0s 13ms/step - loss: 1.2231 - mean_absolute_error: 0.2786 - val_loss: 1.0210 - val_mean_absolute_error: 0.3147
Epoch 53/250
18/18 [==============================] - 0s 8ms/step - loss: 0.3149 - mean_absolute_error: 0.2330 - val_loss: 1.0210 - val_mean_absolute_error: 0.3147
Epoch 54/250
18/18 [==============================] - 0s 8ms/step - loss: 0.3882 - mean_absolute_error: 0.2305 - val_loss: 1.0210 - val_mean_absolute_error: 0.3147
Epoch 55/250
18/18 [==============================] - 0s 7ms/step - loss: 2.8815 - mean_absolute_error: 0.3978 - val_loss: 1.0210 - val_mean_absolute_error: 0.3147
Epoch 56/250
18/18 [==============================] - 0s 4ms/step - loss: 0.3304 - mean_absolute_error: 0.2429 - val_loss: 1.0210 - val_mean_absolute_error: 0.3147
Epoch 57/250
18

Epoch 101/250
18/18 [==============================] - 0s 4ms/step - loss: 1.2556 - mean_absolute_error: 0.2950 - val_loss: 1.0210 - val_mean_absolute_error: 0.3147
Epoch 102/250
18/18 [==============================] - 0s 4ms/step - loss: 2.0360 - mean_absolute_error: 0.3179 - val_loss: 1.0210 - val_mean_absolute_error: 0.3147
Epoch 103/250
18/18 [==============================] - 0s 8ms/step - loss: 0.9757 - mean_absolute_error: 0.2684 - val_loss: 1.0210 - val_mean_absolute_error: 0.3147
Epoch 104/250
18/18 [==============================] - 0s 5ms/step - loss: 1.4796 - mean_absolute_error: 0.3158 - val_loss: 1.0210 - val_mean_absolute_error: 0.3147
Epoch 105/250
18/18 [==============================] - 0s 5ms/step - loss: 0.8741 - mean_absolute_error: 0.2728 - val_loss: 1.0210 - val_mean_absolute_error: 0.3147
Epoch 106/250
18/18 [==============================] - 0s 8ms/step - loss: 0.4753 - mean_absolute_error: 0.2580 - val_loss: 1.0210 - val_mean_absolute_error: 0.3147
Epoch 107/

Epoch 151/250
18/18 [==============================] - 0s 5ms/step - loss: 1.0030 - mean_absolute_error: 0.2792 - val_loss: 1.0210 - val_mean_absolute_error: 0.3147
Epoch 152/250
18/18 [==============================] - 0s 5ms/step - loss: 0.7534 - mean_absolute_error: 0.2834 - val_loss: 1.0210 - val_mean_absolute_error: 0.3147
Epoch 153/250
18/18 [==============================] - 0s 5ms/step - loss: 0.3441 - mean_absolute_error: 0.2499 - val_loss: 1.0210 - val_mean_absolute_error: 0.3147
Epoch 154/250
18/18 [==============================] - 0s 9ms/step - loss: 0.5279 - mean_absolute_error: 0.2676 - val_loss: 1.0210 - val_mean_absolute_error: 0.3147
Epoch 155/250
18/18 [==============================] - 0s 6ms/step - loss: 0.8298 - mean_absolute_error: 0.2542 - val_loss: 1.0210 - val_mean_absolute_error: 0.3147
Epoch 156/250
18/18 [==============================] - 0s 6ms/step - loss: 1.6703 - mean_absolute_error: 0.3076 - val_loss: 1.0210 - val_mean_absolute_error: 0.3147
Epoch 157/

Epoch 250/250
18/18 [==============================] - 0s 7ms/step - loss: 0.5302 - mean_absolute_error: 0.2841 - val_loss: 1.0210 - val_mean_absolute_error: 0.3147


In [93]:
scores = model.evaluate(X, y)
print("%s: %.2f" %  (model.metrics_names[1], scores[1]))# %  *100

17/17 [==============================] - 0s 1ms/step - loss: 1.0000 - mean_absolute_error: 0.2920
mean_absolute_error: 0.29


In [95]:
model.evaluate(X, y)

17/17 [==============================] - 0s 1ms/step - loss: 1.0000 - mean_absolute_error: 0.2920


[1.0000001192092896, 0.291957825422287]

In [96]:
pred_y = model.predict(X)
pred_y = pred_y.flatten()
pd.DataFrame(zip(pred_y,y)).head()

0        1
0  0.0 -0.20202
1  0.0 -0.20202
2  0.0 -0.20202
3  0.0 -0.20202
4  0.0 -0.20202

In [56]:
from sklearn.model_selection import GridSearchCV, KFold
from keras.wrappers.scikit_learn import KerasClassifier,KerasRegressor
from keras.optimizers import Adam
from keras.layers import Dropout

In [57]:
def create_model(learning_rate,dropout_rate,activation_function,init,neuron1,neuron2): #neuron3
    model = Sequential()
    model.add(Dense(neuron1,input_dim = 8,kernel_initializer = init,activation = activation_function))
    model.add(Dropout(dropout_rate))
    model.add(Dense(neuron2,input_dim = neuron1,kernel_initializer = init,activation = activation_function))
    model.add(Dropout(dropout_rate))
    model.add(Dense(1,activation = activation_function))
    
    adam = Adam(learning_rate = learning_rate)
    model.compile(loss = 'mse',optimizer = adam,metrics = 'mean_absolute_error')
    return model

# Create the model

model = KerasRegressor(build_fn = create_model,verbose = 0) #KerasClassifier

# Define the grid search parameters

batch_size = [10,20]
epochs = [10,50]
learning_rate = [0.01,0.1]
dropout_rate = [0.01,0.1]
activation_function = ['LeakyReLU','relu']
init = ['uniform','normal','zero']
neuron1 = [32,64,128,256]
neuron2 = [8,16,32]
# neuron3 = [2,4]

# Make a dictionary of the grid search parameters

param_grids = dict(batch_size = batch_size,epochs = epochs,learning_rate = learning_rate,dropout_rate = dropout_rate,
                   activation_function = activation_function,init = init,neuron1 = neuron1,neuron2 = neuron2) #,neuron3=neuron3

# Build and fit the GridSearchCV

grid = GridSearchCV(estimator = model,param_grid = param_grids,cv = KFold(n_splits=2),verbose = 1,n_jobs=-1)
grid_result = grid.fit(X,y)

Fitting 2 folds for each of 1152 candidates, totalling 2304 fits


[Parallel(n_jobs=-1)]: Using backend LokyBackend with 8 concurrent workers.
[Parallel(n_jobs=-1)]: Done  34 tasks      | elapsed:   57.9s
[Parallel(n_jobs=-1)]: Done 184 tasks      | elapsed:  2.6min
[Parallel(n_jobs=-1)]: Done 434 tasks      | elapsed:  5.5min
[Parallel(n_jobs=-1)]: Done 784 tasks      | elapsed:  9.6min
[Parallel(n_jobs=-1)]: Done 1234 tasks      | elapsed: 14.2min
[Parallel(n_jobs=-1)]: Done 1784 tasks      | elapsed: 20.5min
[Parallel(n_jobs=-1)]: Done 2304 out of 2304 | elapsed: 25.5min finished


In [58]:
# Summarize the results
print('Best : {}, using {}'.format(grid_result.best_score_,grid_result.best_params_))
means = grid_result.cv_results_['mean_test_score']
stds = grid_result.cv_results_['std_test_score']
params = grid_result.cv_results_['params']
for mean, stdev, param in zip(means, stds, params):
  print('{},{} with: {}'.format(mean, stdev, param))

Best : -0.9925381541252136, using {'activation_function': 'LeakyReLU', 'batch_size': 20, 'dropout_rate': 0.1, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 64, 'neuron2': 8}
-1.018142580986023,0.2966034412384033 with: {'activation_function': 'LeakyReLU', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 32, 'neuron2': 8}
-1.0109500885009766,0.2800624370574951 with: {'activation_function': 'LeakyReLU', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 32, 'neuron2': 16}
-1.0180040895938873,0.31061384081840515 with: {'activation_function': 'LeakyReLU', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 32, 'neuron2': 32}
-1.0229385495185852,0.28747648000717163 with: {'activation_function': 'LeakyReLU', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'ne

In [60]:
def create_model(learning_rate,dropout_rate,activation_function,init,neuron1,neuron2): #neuron3
    model = Sequential()
    model.add(Dense(neuron1,input_dim = 8,kernel_initializer = init,activation = activation_function))
    model.add(Dropout(dropout_rate))
    model.add(Dense(neuron2,input_dim = neuron1,kernel_initializer = init,activation = activation_function))
    model.add(Dropout(dropout_rate))
    model.add(Dense(1,activation = activation_function)) #neuron3
    
    adam = Adam(learning_rate = learning_rate)
    model.compile(loss = 'mean_squared_error',optimizer = adam,metrics = 'mean_squared_error')
    return model

# Create the model

model = KerasRegressor(build_fn = create_model,verbose = 0)

# Define the grid search parameters

batch_size = [10,20]
epochs = [10,50]
learning_rate = [0.01,0.1]
dropout_rate = [0.01,0.1]
activation_function = ['Softmax',]
init = ['uniform','normal','zero']
neuron1 = [32,64,128,256]
neuron2 = [16,32,64]
# neuron3 = [2,4]

# Make a dictionary of the grid search parameters

param_grids = dict(batch_size = batch_size,epochs = epochs,learning_rate = learning_rate,dropout_rate = dropout_rate,
                   activation_function = activation_function,init = init,neuron1 = neuron1,neuron2 = neuron2) #,neuron3=neuron3

# Build and fit the GridSearchCV

grid = GridSearchCV(estimator = model,param_grid = param_grids,cv = KFold(n_splits=2),verbose = 1,n_jobs=-1)
grid_result = grid.fit(X,y)

Fitting 2 folds for each of 576 candidates, totalling 1152 fits


[Parallel(n_jobs=-1)]: Using backend LokyBackend with 8 concurrent workers.
[Parallel(n_jobs=-1)]: Done  34 tasks      | elapsed:  1.1min
[Parallel(n_jobs=-1)]: Done 184 tasks      | elapsed:  2.8min
[Parallel(n_jobs=-1)]: Done 434 tasks      | elapsed:  5.7min
[Parallel(n_jobs=-1)]: Done 784 tasks      | elapsed: 10.0min
[Parallel(n_jobs=-1)]: Done 1152 out of 1152 | elapsed: 13.9min finished


In [61]:
# Summarize the results
print('Best : {}, using {}'.format(grid_result.best_score_,grid_result.best_params_))
means = grid_result.cv_results_['mean_test_score']
stds = grid_result.cv_results_['std_test_score']
params = grid_result.cv_results_['params']
for mean, stdev, param in zip(means, stds, params):
  print('{},{} with: {}'.format(mean, stdev, param))

Best : -1.9994363188743591, using {'activation_function': 'Softmax', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 32, 'neuron2': 16}
-1.9994363188743591,0.29142266511917114 with: {'activation_function': 'Softmax', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 32, 'neuron2': 16}
-1.9994363188743591,0.29142266511917114 with: {'activation_function': 'Softmax', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 32, 'neuron2': 32}
-1.9994363188743591,0.29142266511917114 with: {'activation_function': 'Softmax', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 32, 'neuron2': 64}
-1.9994363188743591,0.29142266511917114 with: {'activation_function': 'Softmax', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron

In [62]:
def create_model(learning_rate,dropout_rate,activation_function,init,neuron1,neuron2): #neuron3
    model = Sequential()
    model.add(Dense(neuron1,input_dim = 8,kernel_initializer = init,activation = activation_function))
    model.add(Dropout(dropout_rate))
    model.add(Dense(neuron2,input_dim = neuron1,kernel_initializer = init,activation = activation_function))
    model.add(Dropout(dropout_rate))
    model.add(Dense(1,activation = activation_function)) #neuron3
    
    adam = Adam(learning_rate = learning_rate)
    model.compile(loss = 'mean_squared_error',optimizer = adam,metrics = 'mean_squared_error')
    return model

# Create the model

model = KerasRegressor(build_fn = create_model,verbose = 0)

# Define the grid search parameters

batch_size = [10,20]
epochs = [10,50]
learning_rate = [0.01,0.1]
dropout_rate = [0.01,0.1]
activation_function = ['linear']
init = ['uniform','normal','zero']
neuron1 = [32,64,128,256]
neuron2 = [16,32,64]
# neuron3 = [2,4]

# Make a dictionary of the grid search parameters

param_grids = dict(batch_size = batch_size,epochs = epochs,learning_rate = learning_rate,dropout_rate = dropout_rate,
                   activation_function = activation_function,init = init,neuron1 = neuron1,neuron2 = neuron2) #,neuron3=neuron3

# Build and fit the GridSearchCV

grid = GridSearchCV(estimator = model,param_grid = param_grids,cv = KFold(n_splits=2),verbose = 1,n_jobs=-1)
grid_result = grid.fit(X,y)

Fitting 2 folds for each of 576 candidates, totalling 1152 fits


[Parallel(n_jobs=-1)]: Using backend LokyBackend with 8 concurrent workers.
[Parallel(n_jobs=-1)]: Done  34 tasks      | elapsed:   18.9s
[Parallel(n_jobs=-1)]: Done 184 tasks      | elapsed:  1.8min
[Parallel(n_jobs=-1)]: Done 434 tasks      | elapsed:  4.5min
[Parallel(n_jobs=-1)]: Done 784 tasks      | elapsed:  8.2min
[Parallel(n_jobs=-1)]: Done 1152 out of 1152 | elapsed: 11.7min finished


In [63]:
# Summarize the results
print('Best : {}, using {}'.format(grid_result.best_score_,grid_result.best_params_))
means = grid_result.cv_results_['mean_test_score']
stds = grid_result.cv_results_['std_test_score']
params = grid_result.cv_results_['params']
for mean, stdev, param in zip(means, stds, params):
  print('{},{} with: {}'.format(mean, stdev, param))

Best : -0.9921384751796722, using {'activation_function': 'linear', 'batch_size': 20, 'dropout_rate': 0.1, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 64, 'neuron2': 16}
-1.0173256695270538,0.2844190299510956 with: {'activation_function': 'linear', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 32, 'neuron2': 16}
-0.9976673424243927,0.301351398229599 with: {'activation_function': 'linear', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 32, 'neuron2': 32}
-0.9987383782863617,0.3030288517475128 with: {'activation_function': 'linear', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 32, 'neuron2': 64}
-0.9969885647296906,0.3026023805141449 with: {'activation_function': 'linear', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 64, 'ne

In [66]:
def create_model(learning_rate,dropout_rate,activation_function,init,neuron1,neuron2): #neuron3
    model = Sequential()
    model.add(Dense(neuron1,input_dim = 8,kernel_initializer = init,activation = activation_function))
    model.add(Dropout(dropout_rate))
    model.add(Dense(neuron2,input_dim = neuron1,kernel_initializer = init,activation = activation_function))
    model.add(Dropout(dropout_rate))
    model.add(Dense(1,activation = activation_function)) #neuron3
    
    adam = Adam(learning_rate = learning_rate)
    model.compile(loss = 'mean_squared_error',optimizer = adam,metrics = 'mean_squared_error')
    return model

# Create the model

model = KerasRegressor(build_fn = create_model,verbose = 0)

# Define the grid search parameters

batch_size = [10,20]
epochs = [10,50]
learning_rate = [0.01,0.1]
dropout_rate = [0.01,0.1]
activation_function = ['softplus',]
init = ['uniform','normal','zero']
neuron1 = [32,64,128,256]
neuron2 = [16,32,64]
# neuron3 = [2,4]

# Make a dictionary of the grid search parameters

param_grids = dict(batch_size = batch_size,epochs = epochs,learning_rate = learning_rate,dropout_rate = dropout_rate,
                   activation_function = activation_function,init = init,neuron1 = neuron1,neuron2 = neuron2) #,neuron3=neuron3

# Build and fit the GridSearchCV

grid = GridSearchCV(estimator = model,param_grid = param_grids,cv = KFold(n_splits=2),verbose = 1,n_jobs=-1)
grid_result = grid.fit(X,y)

Fitting 2 folds for each of 576 candidates, totalling 1152 fits


[Parallel(n_jobs=-1)]: Using backend LokyBackend with 8 concurrent workers.
[Parallel(n_jobs=-1)]: Done  34 tasks      | elapsed:   58.7s
[Parallel(n_jobs=-1)]: Done 184 tasks      | elapsed:  2.6min
[Parallel(n_jobs=-1)]: Done 434 tasks      | elapsed:  5.5min
[Parallel(n_jobs=-1)]: Done 784 tasks      | elapsed:  9.5min
[Parallel(n_jobs=-1)]: Done 1152 out of 1152 | elapsed: 13.1min finished


In [67]:
# Summarize the results
print('Best : {}, using {}'.format(grid_result.best_score_,grid_result.best_params_))
means = grid_result.cv_results_['mean_test_score']
stds = grid_result.cv_results_['std_test_score']
params = grid_result.cv_results_['params']
for mean, stdev, param in zip(means, stds, params):
  print('{},{} with: {}'.format(mean, stdev, param))

Best : -0.9949370324611664, using {'activation_function': 'softplus', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'normal', 'learning_rate': 0.01, 'neuron1': 32, 'neuron2': 16}
-0.9958063066005707,0.3080299198627472 with: {'activation_function': 'softplus', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 32, 'neuron2': 16}
-0.9994010329246521,0.3062208294868469 with: {'activation_function': 'softplus', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 32, 'neuron2': 32}
-0.9994074106216431,0.30622315406799316 with: {'activation_function': 'softplus', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 32, 'neuron2': 64}
-0.9993789792060852,0.3062290549278259 with: {'activation_function': 'softplus', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuro

In [68]:
def create_model(learning_rate,dropout_rate,activation_function,init,neuron1,neuron2): #neuron3
    model = Sequential()
    model.add(Dense(neuron1,input_dim = 8,kernel_initializer = init,activation = activation_function))
    model.add(Dropout(dropout_rate))
    model.add(Dense(neuron2,input_dim = neuron1,kernel_initializer = init,activation = activation_function))
    model.add(Dropout(dropout_rate))
    model.add(Dense(1,activation = activation_function)) #neuron3
    
    adam = Adam(learning_rate = learning_rate)
    model.compile(loss = 'mean_squared_error',optimizer = adam,metrics = 'mean_squared_error')
    return model

# Create the model

model = KerasRegressor(build_fn = create_model,verbose = 0)

# Define the grid search parameters

batch_size = [10,20]
epochs = [10,50]
learning_rate = [0.01,0.1]
dropout_rate = [0.01,0.1]
activation_function = ['relu',]
init = ['uniform','normal','zero']
neuron1 = [32,64,128,256]
neuron2 = [16,32,64]
# neuron3 = [2,4]

# Make a dictionary of the grid search parameters

param_grids = dict(batch_size = batch_size,epochs = epochs,learning_rate = learning_rate,dropout_rate = dropout_rate,
                   activation_function = activation_function,init = init,neuron1 = neuron1,neuron2 = neuron2) #,neuron3=neuron3

# Build and fit the GridSearchCV

grid = GridSearchCV(estimator = model,param_grid = param_grids,cv = KFold(n_splits=2),verbose = 1,n_jobs=-1)
grid_result = grid.fit(X,y)

Fitting 2 folds for each of 576 candidates, totalling 1152 fits


[Parallel(n_jobs=-1)]: Using backend LokyBackend with 8 concurrent workers.
[Parallel(n_jobs=-1)]: Done  34 tasks      | elapsed:   19.9s
[Parallel(n_jobs=-1)]: Done 184 tasks      | elapsed:  1.8min
[Parallel(n_jobs=-1)]: Done 434 tasks      | elapsed:  4.6min
[Parallel(n_jobs=-1)]: Done 784 tasks      | elapsed:  8.4min
[Parallel(n_jobs=-1)]: Done 1152 out of 1152 | elapsed: 12.0min finished


In [69]:
# Summarize the results
print('Best : {}, using {}'.format(grid_result.best_score_,grid_result.best_params_))
means = grid_result.cv_results_['mean_test_score']
stds = grid_result.cv_results_['std_test_score']
params = grid_result.cv_results_['params']
for mean, stdev, param in zip(means, stds, params):
  print('{},{} with: {}'.format(mean, stdev, param))

Best : -0.999407559633255, using {'activation_function': 'relu', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 32, 'neuron2': 16}
-0.999407559633255,0.3062227666378021 with: {'activation_function': 'relu', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 32, 'neuron2': 16}
-0.999407559633255,0.3062227666378021 with: {'activation_function': 'relu', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 32, 'neuron2': 32}
-0.999407559633255,0.3062227666378021 with: {'activation_function': 'relu', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 32, 'neuron2': 64}
-0.999407559633255,0.3062227666378021 with: {'activation_function': 'relu', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 64, 'neuron2': 16}
-

In [70]:
def create_model(learning_rate,dropout_rate,activation_function,init,neuron1,neuron2): #neuron3
    model = Sequential()
    model.add(Dense(neuron1,input_dim = 8,kernel_initializer = init,activation = activation_function))
    model.add(Dropout(dropout_rate))
    model.add(Dense(neuron2,input_dim = neuron1,kernel_initializer = init,activation = activation_function))
    model.add(Dropout(dropout_rate))
    model.add(Dense(1,activation = activation_function)) #neuron3
    
    adam = Adam(learning_rate = learning_rate)
    model.compile(loss = 'mean_squared_error',optimizer = adam,metrics = 'mean_squared_error')
    return model

# Create the model

model = KerasRegressor(build_fn = create_model,verbose = 0)

# Define the grid search parameters

batch_size = [10,20]
epochs = [10,50]
learning_rate = [0.01,0.1]
dropout_rate = [0.01,0.1]
activation_function = ['elu',]
init = ['uniform','normal','zero']
neuron1 = [32,64,128,256]
neuron2 = [16,32,64]
# neuron3 = [2,4]

# Make a dictionary of the grid search parameters

param_grids = dict(batch_size = batch_size,epochs = epochs,learning_rate = learning_rate,dropout_rate = dropout_rate,
                   activation_function = activation_function,init = init,neuron1 = neuron1,neuron2 = neuron2) #,neuron3=neuron3

# Build and fit the GridSearchCV

grid = GridSearchCV(estimator = model,param_grid = param_grids,cv = KFold(n_splits=2),verbose = 1,n_jobs=-1)
grid_result = grid.fit(X,y)

Fitting 2 folds for each of 576 candidates, totalling 1152 fits


[Parallel(n_jobs=-1)]: Using backend LokyBackend with 8 concurrent workers.
[Parallel(n_jobs=-1)]: Done  34 tasks      | elapsed:   23.4s
[Parallel(n_jobs=-1)]: Done 184 tasks      | elapsed:  2.7min
[Parallel(n_jobs=-1)]: Done 434 tasks      | elapsed:  6.6min
[Parallel(n_jobs=-1)]: Done 784 tasks      | elapsed: 12.0min
[Parallel(n_jobs=-1)]: Done 1152 out of 1152 | elapsed: 17.1min finished


In [71]:
# Summarize the results
print('Best : {}, using {}'.format(grid_result.best_score_,grid_result.best_params_))
means = grid_result.cv_results_['mean_test_score']
stds = grid_result.cv_results_['std_test_score']
params = grid_result.cv_results_['params']
for mean, stdev, param in zip(means, stds, params):
  print('{},{} with: {}'.format(mean, stdev, param))

Best : -0.9905881583690643, using {'activation_function': 'elu', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'normal', 'learning_rate': 0.01, 'neuron1': 32, 'neuron2': 64}
-1.003146767616272,0.29921483993530273 with: {'activation_function': 'elu', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 32, 'neuron2': 16}
-1.0107168555259705,0.294511616230011 with: {'activation_function': 'elu', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 32, 'neuron2': 32}
-0.9988830387592316,0.3009837567806244 with: {'activation_function': 'elu', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 32, 'neuron2': 64}
-1.015911877155304,0.2930982708930969 with: {'activation_function': 'elu', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 64, 'neuron2': 16}
-1.0

In [72]:
def create_model(learning_rate,dropout_rate,activation_function,init,neuron1,neuron2): #neuron3
    model = Sequential()
    model.add(Dense(neuron1,input_dim = 8,kernel_initializer = init,activation = activation_function))
    model.add(Dropout(dropout_rate))
    model.add(Dense(neuron2,input_dim = neuron1,kernel_initializer = init,activation = activation_function))
    model.add(Dropout(dropout_rate))
    model.add(Dense(1,activation = activation_function)) #neuron3
    
    adam = Adam(learning_rate = learning_rate)
    model.compile(loss = 'mean_squared_error',optimizer = adam,metrics = 'mean_squared_error')
    return model

# Create the model

model = KerasRegressor(build_fn = create_model,verbose = 0)

# Define the grid search parameters

batch_size = [10,20]
epochs = [10,50]
learning_rate = [0.01,0.1]
dropout_rate = [0.01,0.1]
activation_function = ['selu',]
init = ['uniform','normal','zero']
neuron1 = [32,64,128,256]
neuron2 = [16,32,64]
# neuron3 = [2,4]

# Make a dictionary of the grid search parameters

param_grids = dict(batch_size = batch_size,epochs = epochs,learning_rate = learning_rate,dropout_rate = dropout_rate,
                   activation_function = activation_function,init = init,neuron1 = neuron1,neuron2 = neuron2) #,neuron3=neuron3

# Build and fit the GridSearchCV

grid = GridSearchCV(estimator = model,param_grid = param_grids,cv = KFold(n_splits=2),verbose = 1,n_jobs=-1)
grid_result = grid.fit(X,y)

Fitting 2 folds for each of 576 candidates, totalling 1152 fits


[Parallel(n_jobs=-1)]: Using backend LokyBackend with 8 concurrent workers.
[Parallel(n_jobs=-1)]: Done  34 tasks      | elapsed:   27.5s
[Parallel(n_jobs=-1)]: Done 184 tasks      | elapsed:  2.6min
[Parallel(n_jobs=-1)]: Done 434 tasks      | elapsed:  6.5min
[Parallel(n_jobs=-1)]: Done 784 tasks      | elapsed: 11.9min
[Parallel(n_jobs=-1)]: Done 1152 out of 1152 | elapsed: 17.0min finished


In [73]:
# Summarize the results
print('Best : {}, using {}'.format(grid_result.best_score_,grid_result.best_params_))
means = grid_result.cv_results_['mean_test_score']
stds = grid_result.cv_results_['std_test_score']
params = grid_result.cv_results_['params']
for mean, stdev, param in zip(means, stds, params):
  print('{},{} with: {}'.format(mean, stdev, param))

Best : -0.9956141710281372, using {'activation_function': 'selu', 'batch_size': 20, 'dropout_rate': 0.1, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 128, 'neuron2': 64}
-0.9980852007865906,0.3018508553504944 with: {'activation_function': 'selu', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 32, 'neuron2': 16}
-1.0044948756694794,0.29636791348457336 with: {'activation_function': 'selu', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 32, 'neuron2': 32}
-1.0176370441913605,0.2821703851222992 with: {'activation_function': 'selu', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 32, 'neuron2': 64}
-1.0078608691692352,0.2933538258075714 with: {'activation_function': 'selu', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 64, 'neuron2':

In [74]:
def create_model(learning_rate,dropout_rate,activation_function,init,neuron1,neuron2): #neuron3
    model = Sequential()
    model.add(Dense(neuron1,input_dim = 8,kernel_initializer = init,activation = activation_function))
    model.add(Dropout(dropout_rate))
    model.add(Dense(neuron2,input_dim = neuron1,kernel_initializer = init,activation = activation_function))
    model.add(Dropout(dropout_rate))
    model.add(Dense(1,activation = activation_function)) #neuron3
    
    adam = Adam(learning_rate = learning_rate)
    model.compile(loss = 'mean_squared_error',optimizer = adam,metrics = 'mean_squared_error')
    return model

# Create the model

model = KerasRegressor(build_fn = create_model,verbose = 0)

# Define the grid search parameters

batch_size = [10,20]
epochs = [10,50]
learning_rate = [0.01,0.1]
dropout_rate = [0.01,0.1]
activation_function = ['softplus',]
init = ['uniform','normal','zero']
neuron1 = [32,64,128,256]
neuron2 = [16,32,64]
# neuron3 = [2,4]

# Make a dictionary of the grid search parameters

param_grids = dict(batch_size = batch_size,epochs = epochs,learning_rate = learning_rate,dropout_rate = dropout_rate,
                   activation_function = activation_function,init = init,neuron1 = neuron1,neuron2 = neuron2) #,neuron3=neuron3

# Build and fit the GridSearchCV

grid = GridSearchCV(estimator = model,param_grid = param_grids,cv = KFold(n_splits=2),verbose = 1,n_jobs=-1)
grid_result = grid.fit(X,y)

Fitting 2 folds for each of 576 candidates, totalling 1152 fits


[Parallel(n_jobs=-1)]: Using backend LokyBackend with 8 concurrent workers.
[Parallel(n_jobs=-1)]: Done  34 tasks      | elapsed:   27.8s
[Parallel(n_jobs=-1)]: Done 184 tasks      | elapsed:  2.6min
[Parallel(n_jobs=-1)]: Done 434 tasks      | elapsed:  6.6min
[Parallel(n_jobs=-1)]: Done 784 tasks      | elapsed: 12.2min
[Parallel(n_jobs=-1)]: Done 1152 out of 1152 | elapsed: 17.3min finished


In [75]:
# Summarize the results
print('Best : {}, using {}'.format(grid_result.best_score_,grid_result.best_params_))
means = grid_result.cv_results_['mean_test_score']
stds = grid_result.cv_results_['std_test_score']
params = grid_result.cv_results_['params']
for mean, stdev, param in zip(means, stds, params):
  print('{},{} with: {}'.format(mean, stdev, param))

Best : -0.9937363266944885, using {'activation_function': 'softplus', 'batch_size': 20, 'dropout_rate': 0.01, 'epochs': 50, 'init': 'normal', 'learning_rate': 0.01, 'neuron1': 64, 'neuron2': 32}
-0.9956945180892944,0.30659329891204834 with: {'activation_function': 'softplus', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 32, 'neuron2': 16}
-0.997444212436676,0.304298460483551 with: {'activation_function': 'softplus', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 32, 'neuron2': 32}
-0.9993766248226166,0.30625370144844055 with: {'activation_function': 'softplus', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 32, 'neuron2': 64}
-0.9991660416126251,0.30598267912864685 with: {'activation_function': 'softplus', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuro

In [76]:
def create_model(learning_rate,dropout_rate,activation_function,init,neuron1,neuron2): #neuron3
    model = Sequential()
    model.add(Dense(neuron1,input_dim = 8,kernel_initializer = init,activation = activation_function))
    model.add(Dropout(dropout_rate))
    model.add(Dense(neuron2,input_dim = neuron1,kernel_initializer = init,activation = activation_function))
    model.add(Dropout(dropout_rate))
    model.add(Dense(1,activation = activation_function)) #neuron3
    
    adam = Adam(learning_rate = learning_rate)
    model.compile(loss = 'mean_squared_error',optimizer = adam,metrics = 'mean_squared_error')
    return model

# Create the model

model = KerasRegressor(build_fn = create_model,verbose = 0)

# Define the grid search parameters

batch_size = [10,20]
epochs = [10,50]
learning_rate = [0.01,0.1]
dropout_rate = [0.01,0.1]
activation_function = ['exponential',]
init = ['uniform','normal','zero']
neuron1 = [32,64,128,256]
neuron2 = [16,32,64]
# neuron3 = [2,4]

# Make a dictionary of the grid search parameters

param_grids = dict(batch_size = batch_size,epochs = epochs,learning_rate = learning_rate,dropout_rate = dropout_rate,
                   activation_function = activation_function,init = init,neuron1 = neuron1,neuron2 = neuron2) #,neuron3=neuron3

# Build and fit the GridSearchCV

grid = GridSearchCV(estimator = model,param_grid = param_grids,cv = KFold(n_splits=2),verbose = 1,n_jobs=-1)
grid_result = grid.fit(X,y)

Fitting 2 folds for each of 576 candidates, totalling 1152 fits


[Parallel(n_jobs=-1)]: Using backend LokyBackend with 8 concurrent workers.
[Parallel(n_jobs=-1)]: Done  34 tasks      | elapsed:   27.8s
[Parallel(n_jobs=-1)]: Done 184 tasks      | elapsed:  2.6min
[Parallel(n_jobs=-1)]: Done 434 tasks      | elapsed:  6.5min
[Parallel(n_jobs=-1)]: Done 784 tasks      | elapsed: 12.0min
[Parallel(n_jobs=-1)]: Done 1152 out of 1152 | elapsed: 17.1min finished


In [77]:
# Summarize the results
print('Best : {}, using {}'.format(grid_result.best_score_,grid_result.best_params_))
means = grid_result.cv_results_['mean_test_score']
stds = grid_result.cv_results_['std_test_score']
params = grid_result.cv_results_['params']
for mean, stdev, param in zip(means, stds, params):
  print('{},{} with: {}'.format(mean, stdev, param))

Best : -0.9993249177932739, using {'activation_function': 'exponential', 'batch_size': 20, 'dropout_rate': 0.1, 'epochs': 50, 'init': 'zero', 'learning_rate': 0.01, 'neuron1': 32, 'neuron2': 16}
-0.999407559633255,0.3062227666378021 with: {'activation_function': 'exponential', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 32, 'neuron2': 16}
-0.999407559633255,0.3062227666378021 with: {'activation_function': 'exponential', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 32, 'neuron2': 32}
-0.999407559633255,0.3062227666378021 with: {'activation_function': 'exponential', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 32, 'neuron2': 64}
-0.999407559633255,0.3062227666378021 with: {'activation_function': 'exponential', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01,

In [81]:
from keras.optimizers import RMSprop 

In [83]:
def create_model(learning_rate,dropout_rate,activation_function,init,neuron1,neuron2): #neuron3
    model = Sequential()
    model.add(Dense(neuron1,input_dim = 8,kernel_initializer = init,activation = activation_function))
    model.add(Dropout(dropout_rate))
    model.add(Dense(neuron2,input_dim = neuron1,kernel_initializer = init,activation = activation_function))
    model.add(Dropout(dropout_rate))
#     model.add(Dense(1,activation = activation_function)) #neuron3
    
#     adam = Adam(learning_rate = learning_rate)
    model.compile(loss = 'mean_squared_error',optimizer = RMSprop(),metrics = ['mean_squared_error'])
    return model

# Create the model

model = KerasRegressor(build_fn = create_model,verbose = 0)

# Define the grid search parameters

batch_size = [10,20]
epochs = [10,50]
learning_rate = [0.01,0.1]
dropout_rate = [0.01,0.1]
activation_function = ['softplus',]
init = ['uniform','normal','zero']
neuron1 = [32,64,128,256]
neuron2 = [16,32,64]
# neuron3 = [2,4]

# Make a dictionary of the grid search parameters

param_grids = dict(batch_size = batch_size,epochs = epochs,learning_rate = learning_rate,dropout_rate = dropout_rate,
                   activation_function = activation_function,init = init,neuron1 = neuron1,neuron2 = neuron2) #,neuron3=neuron3

# Build and fit the GridSearchCV

grid = GridSearchCV(estimator = model,param_grid = param_grids,cv = KFold(n_splits=2),verbose = 1,n_jobs=-1)
grid_result = grid.fit(X,y)

Fitting 2 folds for each of 576 candidates, totalling 1152 fits


[Parallel(n_jobs=-1)]: Using backend LokyBackend with 8 concurrent workers.
[Parallel(n_jobs=-1)]: Done  34 tasks      | elapsed:   31.7s
[Parallel(n_jobs=-1)]: Done 184 tasks      | elapsed:  2.6min
[Parallel(n_jobs=-1)]: Done 434 tasks      | elapsed:  6.3min
[Parallel(n_jobs=-1)]: Done 784 tasks      | elapsed: 11.6min
[Parallel(n_jobs=-1)]: Done 1152 out of 1152 | elapsed: 16.5min finished


In [84]:
# Summarize the results
print('Best : {}, using {}'.format(grid_result.best_score_,grid_result.best_params_))
means = grid_result.cv_results_['mean_test_score']
stds = grid_result.cv_results_['std_test_score']
params = grid_result.cv_results_['params']
for mean, stdev, param in zip(means, stds, params):
  print('{},{} with: {}'.format(mean, stdev, param))

Best : -0.9978061616420746, using {'activation_function': 'softplus', 'batch_size': 20, 'dropout_rate': 0.1, 'epochs': 50, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 128, 'neuron2': 16}
-1.0012057423591614,0.30482596158981323 with: {'activation_function': 'softplus', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 32, 'neuron2': 16}
-1.0017306804656982,0.30452466011047363 with: {'activation_function': 'softplus', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 32, 'neuron2': 32}
-1.0013195872306824,0.3048756718635559 with: {'activation_function': 'softplus', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuron1': 32, 'neuron2': 64}
-0.9989053606987,0.30613797903060913 with: {'activation_function': 'softplus', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'uniform', 'learning_rate': 0.01, 'neuro

In [89]:
pd.DataFrame(grid_result.cv_results_).head(10).sort_values(by='rank_test_score')

mean_fit_time  std_fit_time  mean_score_time  std_score_time  \
6       3.786662      0.008191         1.622143        0.008901   
8       3.900193      0.037943         1.236232        0.016992   
7       4.329127      0.130391         1.228783        0.182875   
3       9.335920      0.019701         1.709111        0.010077   
5       3.744994      0.017182         1.598048        0.013460   
4       4.262694      0.044386         1.168496        0.020861   
9       3.980363      0.012297         1.197685        0.005201   
0       9.354656      0.011864         1.743579        0.028951   
2       9.724867      0.125232         1.735500        0.028221   
1       9.338367      0.012774         1.710052        0.003119   

  param_activation_function param_batch_size param_dropout_rate param_epochs  \
6                  softplus               10               0.01           10   
8                  softplus               10               0.01           10   
7                  softplus               10               0.01           10   
3                  softplus               10               0.01           10   
5                  softplus               10               0.01           10   
4                  softplus               10               0.01           10   
9                  softplus               10               0.01           10   
0                  softplus               10               0.01           10   
2                  softplus               10               0.01           10   
1                  softplus               10               0.01           10   

  param_init param_learning_rate param_neuron1 param_neuron2  \
6    uniform                0.01           128            16   
8    uniform                0.01           128            64   
7    uniform                0.01           128            32   
3    uniform                0.01            64            16   
5    uniform                0.01            64            64   
4    uniform                0.01            64            32   
9    uniform                0.01           256            16   
0    uniform                0.01            32            16   
2    uniform                0.01            32            64   
1    uniform                0.01            32            32   

                                              params  split0_test_score  \
6  {'activation_function': 'softplus', 'batch_siz...          -1.305321   
8  {'activation_function': 'softplus', 'batch_siz...          -1.305297   
7  {'activation_function': 'softplus', 'batch_siz...          -1.305284   
3  {'activation_function': 'softplus', 'batch_siz...          -1.305043   
5  {'activation_function': 'softplus', 'batch_siz...          -1.305030   
4  {'activation_function': 'softplus', 'batch_siz...          -1.305163   
9  {'activation_function': 'softplus', 'batch_siz...          -1.305497   
0  {'activation_function': 'softplus', 'batch_siz...          -1.306032   
2  {'activation_function': 'softplus', 'batch_siz...          -1.306195   
1  {'activation_function': 'softplus', 'batch_siz...          -1.306255   

   split1_test_score  mean_test_score  std_test_score  rank_test_score  
6          -0.692161        -0.998741        0.306580              196  
8          -0.692224        -0.998761        0.306537              202  
7          -0.692295        -0.998789        0.306494              213  
3          -0.692767        -0.998905        0.306138              260  
5          -0.692887        -0.998958        0.306072              288  
4          -0.692773        -0.998968        0.306195              293  
9          -0.692505        -0.999001        0.306496              313  
0          -0.696380        -1.001206        0.304826              475  
2          -0.696444        -1.001320        0.304876              482  
1          -0.697206        -1.001731        0.304525              528

In [97]:
from sklearn.model_selection import train_test_split

In [98]:
Xtrain,Xtest,ytrain,ytest = train_test_split(X,y,test_size = .2, random_state = 42)

In [99]:
Xtrain.shape,Xtest.shape,ytrain.shape,ytest.shape

((413, 8), (104, 8), (413,), (104,))

Best : -0.9905881583690643, using {'activation_function': 'elu', 'batch_size': 10, 'dropout_rate': 0.01, 'epochs': 10, 'init': 'normal', 'learning_rate': 0.01, 'neuron1': 32, 'neuron2': 64}

In [104]:
model = Sequential()
model.add(Dense(32, input_dim=8, kernel_initializer = 'normal', activation='elu'))
model.add(Dropout(0.01))
model.add(Dense(64, kernel_initializer = 'normal', activation = 'elu'))
model.add(Dropout(0.01))
model.add(Dense(1, activation = 'elu'))

In [105]:
adam = Adam(learning_rate = 0.01)

In [107]:
model.compile(loss='mse', optimizer='adam', metrics='mean_absolute_error' ) 

In [103]:
model.fit(Xtrain, ytrain, validation_split=0.33, epochs=10, batch_size=10,)

Epoch 1/10
35/35 [==============================] - 4s 25ms/step - loss: 0.8430 - mean_absolute_error: 0.2603 - val_loss: 1.0090 - val_mean_absolute_error: 0.3457
Epoch 2/10
35/35 [==============================] - 0s 4ms/step - loss: 0.9525 - mean_absolute_error: 0.2497 - val_loss: 1.0080 - val_mean_absolute_error: 0.3488
Epoch 3/10
35/35 [==============================] - 0s 3ms/step - loss: 0.5594 - mean_absolute_error: 0.2333 - val_loss: 1.0099 - val_mean_absolute_error: 0.3630
Epoch 4/10
35/35 [==============================] - 0s 5ms/step - loss: 0.2974 - mean_absolute_error: 0.2382 - val_loss: 1.0095 - val_mean_absolute_error: 0.3442
Epoch 5/10
35/35 [==============================] - 0s 8ms/step - loss: 0.8368 - mean_absolute_error: 0.2888 - val_loss: 1.0088 - val_mean_absolute_error: 0.3766
Epoch 6/10
35/35 [==============================] - 0s 7ms/step - loss: 0.4632 - mean_absolute_error: 0.2300 - val_loss: 1.0120 - val_mean_absolute_error: 0.3682
Epoch 7/10
35/35 [=========

In [108]:
model.evaluate(Xtest,ytest)

4/4 [==============================] - 1s 8ms/step - loss: 2.7137 - mean_absolute_error: 0.3628


[2.9226155281066895, 0.39488130807876587]

In [109]:
predict = model.predict(Xtest)
print(predict)
print(predict.flatten())

[[-0.00888449]
 [ 0.02228732]
 [ 0.00070385]
 [-0.00848585]
 [ 0.00539306]
 [ 0.01123303]
 [ 0.01107955]
 [-0.00862098]
 [-0.00233567]
 [-0.0084033 ]
 [ 0.00214607]
 [-0.01600242]
 [ 0.00615832]
 [-0.00167769]
 [ 0.00592544]
 [-0.03446281]
 [-0.02228123]
 [-0.00330502]
 [ 0.0040004 ]
 [ 0.01006513]
 [ 0.01531565]
 [-0.00565237]
 [ 0.00309161]
 [ 0.00437317]
 [-0.0073024 ]
 [ 0.00357827]
 [ 0.01156334]
 [-0.00291538]
 [-0.01803571]
 [-0.0275895 ]
 [ 0.0112979 ]
 [-0.00368023]
 [-0.00970542]
 [ 0.00221652]
 [ 0.03061597]
 [ 0.02347659]
 [-0.0177049 ]
 [ 0.00831295]
 [-0.00629157]
 [-0.00893575]
 [ 0.00219254]
 [ 0.00437317]
 [-0.02616495]
 [ 0.00301992]
 [ 0.00835655]
 [ 0.01191591]
 [-0.00792497]
 [ 0.02662205]
 [-0.00114459]
 [ 0.01268106]
 [-0.00882834]
 [ 0.00897962]
 [ 0.00381104]
 [-0.00434285]
 [ 0.00500377]
 [ 0.02675714]
 [-0.02643323]
 [ 0.00342291]
 [-0.0230152 ]
 [-0.01906198]
 [-0.01884985]
 [ 0.00238191]
 [ 0.00644069]
 [-0.00970662]
 [-0.01398337]
 [-0.00957626]
 [-0.01183

In [110]:
pd.DataFrame(zip(predict,ytest))

0         1
0     [-0.00888449] -0.202020
1     [0.022287317] -0.202020
2    [0.0007038489] -0.182678
3    [-0.008485854] -0.179062
4     [0.005393062] -0.171357
..              ...       ...
99   [0.0074474383] -0.162551
100  [-0.020704746] -0.117421
101   [0.032148145] -0.167897
102  [-0.009895265]  0.651674
103  [0.0076097995]  0.338281

[104 rows x 2 columns]